core/src/player.rs

```diff
@@ -179,7 +179,9 @@ pub struct Player {
 
     gc_arena: GcArena,
 
+    /// Current frame rate.
     frame_rate: f64,
+    root_frame_rate: f64,
```

Comment:
This property doesn't need to be cached.; the root movie's declared frame rate can be accessed as `player.swf.frame_rate()`.

Reply:
In context, player is defined as
pub player: Option<Weak<Mutex<Player>>>
I don't know much about Rust, but is accessing context.player.swf.frame_rate() not more expensive (handling the locks) than simply caching it (4 bytes of storage)?